airship/kubeadm-airship-gap-analysis.md

```diff
@@ -13,6 +13,59 @@ Jira Issues:
 Jira Issues:
 - [CA rotation](https://airship.atlassian.net/browse/AIR-138)
 
+ 
+## Use pre-generated certs and user provided CA
+
+This is possible with Kubeadm. We can provide CA certs only or we can provide the whole list of certificates in ```/etc/kubernetes/pki``` . In case we provide CA cert only, kubeadm will generate the other certificates using the CA cert. In the latter case, kubeadm will take the pre-generated certs and avoid generating new ones. The placement of the certs can be done using a script.
+
+In case of CABPK following certificates and keys can be specified by the user:
+```
+cluster-ca-cert
+cluster-ca-key
+    
+etcd-ca-cert  
+etcd-ca-key
+    
+front-proxy-ca-cert
+front-proxy-ca-key
+
+service-account-private-key
+service-account-public-key
+```
+
+## Rotating CA 
+
+Kubeadm currently has an option to renew the CA certificates manually, based on a new CA certificate placed using the method described in previous comment. It is however an alpha version command - 
+
+```kubeadm alpha certs renew```
+
+Our initial experiments with this command however suggest that it has some bugs.(Test bed: Simple Kinder based cluster with one control plane node and kubeadm version 1.15). For example, although we command to renew all the certificates based on the new certificate, the admin.conf does not get updated properly. It is still using the old CA certificate and client certificates. 
+
+In case of HA cluster, the certificate renewal has to be done separately on each control plane node.
+
+
+## Control over certificate validity and rotation
+
+The validity of the certificates can be set while generating CSRs and that can be done in Kubeadm. 
```

Comment:
remove this
 and that can be done in Kubeadm.

Reply:
Done